contracts/test/nativeTransfer.mjs

```diff
@@ -0,0 +1,182 @@
+import hardhat from "hardhat"
+import chai from "chai"
+import {getGasPrice, getTxReceipt} from "../utils/tx.js"
+
+import {isGwMainnetV1} from "../utils/network.js"
+
+const {ethers} = hardhat
+const {expect} = chai
+const {BigNumber} = ethers
+
+let gasPrice, EOA0, EOA1, newEOA0, CA0
+
+if (!isGwMainnetV1()) {
+    gasPrice = await getGasPrice(ethers.provider)
+    EOA0 = (await ethers.getSigners())[0].address
+    EOA1 = (await ethers.getSigners())[1].address
+    newEOA0 = ethers.Wallet.createRandom().address
+    let baseFallbackReceive = await ethers.getContractFactory("baseFallbackReceive")
+    let contract = await baseFallbackReceive.deploy()
+    await contract.deployed()
+    CA0 = contract.address
+}
+
+describe("transfer success", function () {
+    if (isGwMainnetV1()) {
+        return;
+    }
+
+    let tests = [
+        {name: "to EOA", from: EOA0, to: EOA1, value: "0x1", expectGasUsed: "21000"},
+        {name: "to EOA tx.data is not null", from: EOA0, to: EOA1, value: "0x1", data: "0x12", expectGasUsed: "21016"},
+        {name: "to itself", from: EOA0, to: EOA0, value: "0x10", expectGasUsed: "21000"},
+        {name: "transfer 0", from: EOA0, to: EOA1, value: "0x0", expectGasUsed: "21000"},
+        {name: "transfer big value", from: EOA0, to: EOA1, value: "0x8ac7230489e80000", expectGasUsed: "21000"},
+        {name: "to new EOA", from: EOA0, to: newEOA0, value: "0x100", expectGasUsed: "46000"},
+        {name: "to CA", from: EOA0, to: CA0, value: "0x200", expectGasUsed: "21033"},
+        {name: "to CA tx.data is not null", from: EOA0, to: CA0, data: "0x12", value: "0x300", expectGasUsed: "21050"},
+    ]
+
+    for (let i = 0; i < tests.length; i++) {
+        let test = tests[i]
+        it(test.name, async () => {
+            let from_balance = await ethers.provider.getBalance(test.from)
+            let to_balance = await ethers.provider.getBalance(test.to)
+            console.log('before transfer from balance:%s to balance:%s', from_balance, to_balance)
+            let response = await tranfer(test.from, test.to, test.value, test.data)
+            let estimatedGas = await estGas(test.from, test.to, test.value, test.data)
+            let from_balance_sent = await ethers.provider.getBalance(test.from)
```

Comment:
For atomic async operations, we've had issues with multiple tests running in parallel.
Say if multiple `native token` tests are running in parallel:
1. There are workflow A and B
2. A and B uses the same set of accounts for testing
3. When A completed `await transfer()`, B just start to request for `await transfer()`
4. When B completed `await transfer()`, A just start to request for `await ethers.provider.getBalance()`
5. Since global state was changed by `await transfer()` in workflow B, the `from_balance.sub(from_balance_sent)` in A will not match to the expected result `response.gasUsed.mul(gasPrice)`

Can @classicalliu and @Flouse please take a look?

Reply:
The current  case has no problem with serial execution. If cases need to be executed in parallel, the case need to be adjusted.

Reply:
> For atomic async operations, we've had issues with multiple tests running in parallel. Say if multiple `native token` tests are running in parallel:
> 
> 1. There are workflow A and B
> 2. A and B uses the same set of accounts for testing
> 3. When A completed `await transfer()`, B just start to request for `await transfer()`
> 4. When B completed `await transfer()`, A just start to request for `await ethers.provider.getBalance()`
> 5. Since global state was changed by `await transfer()` in workflow B, the result of `from_balance.sub(from_balance_sent)` in A will not match to the expected `response.gasUsed.mul(gasPrice)`
> 
> Can @classicalliu and @Flouse please take a look?

I think so , cases may fail when concurrent

Reply:
I made some adjustments, please help me review it again.@ShookLyngs @classicalliu
Account avlAccount with a balance transfers a fixed amount to randomly generated accounts EOA0 and EOA1 to avoid this problem.

hardhat.config.js
```ts
const PRIVATE_KEY0 = ethers.Wallet.createRandom().privateKey
const PRIVATE_KEY1 = ethers.Wallet.createRandom().privateKey

    gw_alphanet_v1: { // for internal testing
      accounts: [`0x${TEST_PK1}`, `0x${TEST_PK2}`,`${PRIVATE_KEY0}`, `${PRIVATE_KEY1}`],
    },

```
nativeTransfer.mjs
```ts
  const signers = await ethers.getSigners();
  avlAccount = signers[0].address;
  EOA0 = signers[signers.length - 1].address;
  EOA1 = signers[signers.length - 2].address;
  await transfer(avlAccount, EOA0, getValidHex(BigNumber.from("1000000000100000000").mul(gasPrice).toHexString()));
  await transfer(avlAccount, EOA1, getValidHex(BigNumber.from("1").mul(gasPrice).toHexString()));
```

Reply:
I didn't follow this part of changes, and reviewing back I think it might be better if we put random accounts into where they are actually used.